subsys/usb/device_next/usbd_core.c

```diff
@@ -199,6 +201,23 @@ int usbd_device_init_core(struct usbd_contex *const uds_ctx)
 
 int usbd_device_shutdown_core(struct usbd_contex *const uds_ctx)
 {
+	struct usbd_config_node *cfg_nd;
+	int ret;
+
+	SYS_SLIST_FOR_EACH_CONTAINER(&uds_ctx->configs, cfg_nd, node) {
+		uint8_t cfg_value = usbd_config_get_value(cfg_nd);
+
+		ret = usbd_class_remove_all(uds_ctx, cfg_value);
```

Comment:
I think the commits are overly separated. I would prefer if the helpers were added and used in the same commit.

Reply:
They actually are, https://github.com/zephyrproject-rtos/zephyr/pull/58187/commits/6e1fdff22981fdcbcdec6fe91d57df7e1b14f59d. Do you think I should squash it with the last commit?

Reply:
I think one commit for the helpers + use helpers, keep the last commit separate.